File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    le = len(board)
    data = []
    for i in range(le):
        data.append([])
        for j in range(le-1, -1, -1):
            if board[j][i] != 0:
                data[i].append(board[j][i])

    stack = []
    res = 0
    for m in moves:
        if len(data[m-1]) > 0:
            tmp = data[m-1].pop()
            if stack:
                if stack[-1] == tmp: 
                    stack.pop()
                    res += 1
                else:
                    stack.append(tmp)
            else:
                stack.append(tmp)
            
        
    return res*2

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> board, vector<int> moves) {
    int answer = 0; int n = board.size();
    vector<int> v;
    for(int m: moves){
        m -= 1;
        for(int y = 0; y < n; y++){
            if(board[y][m] > 0) {
                int tmp = board[y][m];
                board[y][m] = 0;
                if(!v.empty() && v.back() == tmp) {
                    v.pop_back();
                    answer+=2;
                }
                else {
                    v.push_back(tmp);
                }
                break;
            }
        }
    }
    
    return answer;
}